kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/catalogManaged/CatalogManagedE2EReadSuite.scala

```diff
@@ -293,4 +300,102 @@ class CatalogManagedE2EReadSuite extends AnyFunSuite with TestUtilsWithTableMana
         expectedFileList)
     }
   }
+
+  // We test this in the unit tests as well, but since those use the withProtocolAndMetadata API
+  // we also test it here with a real table where we load the P&M from the log
+  test("reading a catalogManaged table without providing maxCatalogVersion fails") {
+    withCatalogOwnedPreviewTestTable { (tablePath, parsedLogData) =>
+      // With logData
+      intercept[IllegalArgumentException] {
+        TableManager
+          .loadSnapshot(tablePath)
+          .withLogData(parsedLogData.asJava)
+          .build(defaultEngine)
+      }
+      // Without logData (and with time-travel-version)
+      intercept[IllegalArgumentException] {
+        TableManager
+          .loadSnapshot(tablePath)
+          .atVersion(0)
+          .build(defaultEngine)
+      }
+    }
+  }
+
+  test("reading a file-system managed table and providing maxCatalogVersion fails") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      // Create a basic file-system managed table
+      createEmptyTable(tablePath = tablePath, schema = testSchema)
+      // Try to read it and provide the maxCatalogVersion
+      intercept[IllegalArgumentException] {
+        TableManager
+          .loadSnapshot(tablePath)
+          .withMaxCatalogVersion(0)
+          .build(engine)
+      }
+    }
+  }
+
+  test("for latest queries we do not load past the maxRatifiedVersion even if " +
+    "later versions exist on the file-system") {
+    withTempDir { tempDir =>
+      withCatalogOwnedPreviewTestTable { (resourceTablePath, resourceLogData) =>
+        // Copy the catalog-owned-preview test resource table to the temp directory
+        org.apache.commons.io.FileUtils.copyDirectory(
+          new java.io.File(resourceTablePath),
+          tempDir)
+        // "Publish" v1 and v2 (we do both to maintain ordered backfill)
+        val deltaLogPath = new Path(tempDir.getPath, "_delta_log")
+        val stagedCommitPath = new Path(deltaLogPath, "_staged_commits")
+        resourceLogData.foreach { stagedCommit =>
+          val stagedCommitFile = new java.io.File(
+            stagedCommitPath.toString,
+            new Path(stagedCommit.getFileStatus.getPath).getName)
+          val publishedCommitFile = new java.io.File(
+            FileNames.deltaFile(deltaLogPath.toString, stagedCommit.getVersion))
+          org.apache.commons.io.FileUtils.copyFile(stagedCommitFile, publishedCommitFile)
+        }
+
+        // Try to read the table with maxCatalogVersion = 1 (no parsedLogData)
```

Comment:
Can we start at version 0? Or, better yet, could we have a for loop for each allowed version? i.e. we can read versions 0, 1, 2?